jiant/tasks/lib/abductive_nli.py

```diff
@@ -0,0 +1,214 @@
+import numpy as np
+import torch
+from dataclasses import dataclass
+from typing import List
+
+from jiant.tasks.core import (
+    BaseExample,
+    BaseTokenizedExample,
+    BaseDataRow,
+    BatchMixin,
+    Task,
+    TaskTypes,
+)
+from jiant.tasks.lib.templates.shared import (
+    labels_to_bimap,
+    add_cls_token,
+    create_input_set_from_tokens_and_segments,
+)
+from jiant.tasks.utils import truncate_sequences
+from jiant.utils.python.io import read_json_lines
+
+
+@dataclass
+class Example(BaseExample):
+    guid: str
+    input_obs1: str
+    input_hyp1: str
+    input_hyp2: str
+    input_obs2: str
+    label: str
+
+    def tokenize(self, tokenizer):
+        return TokenizedExample(
+            guid=self.guid,
+            input_obs1=tokenizer.tokenize(self.input_obs1),
+            input_hyp1=tokenizer.tokenize(self.input_hyp1),
+            input_hyp2=tokenizer.tokenize(self.input_hyp2),
+            input_obs2=tokenizer.tokenize(self.input_obs2),
+            label_id=AnliTask.LABEL_BIMAP.a[self.label],
```

Comment:
Capturing note here for future reference: looks like LABEL_BIMAP value (which is the return value from call to `labels_to_bimap()`) is a [tuple of Dicts](https://github.com/jiant-dev/jiant/blob/e49bc50cfddf9efdb6061fff2f21b8750abeced3/jiant/tasks/lib/templates/shared.py#L335), i.e., this will raise `AttributeError: 'tuple' object has no attribute 'a'`.

Reply:
(☝️ this came up in the while writing a test for WSC)

Reply:
Ah, this was something I changed when porting over from `nlpr`. I'll update the PR